addons/product_expiry/tests/test_generate_serial_numbers.py

```diff
@@ -163,6 +163,49 @@ def test_set_multiple_lot_name_with_expiration_date_04_written_months(self):
             self.assertEqual(move_line.quantity, 1)
             self.assertEqual(move_line.expiration_date, list_lot_and_qty[i]["datetime"])
 
+    @freeze_time('2025-9-13')
+    def test_set_multiple_lot_name_with_expiration_date_05_import_dates(self):
+        """ When importing lot names, quantities and expiration dates, make sure the date is not
+        replaced by the computed date and if there's no date, it takes the computed date. """
+        product_lot = self.env['product.product'].create({
+            'name': 'Tracked by Lot Numbers',
+            'tracking': 'lot',
+            'is_storable': True,
+            'use_expiration_date': True,
+            'expiration_time': 10,
+        })
```

Comment:
Would you mind putting that in `setUpClass()`? Kinda irritating that it's redefined in *every single test* 😬